core/store/src/trie/insert_delete.rs

```diff
@@ -432,86 +435,84 @@ impl Trie {
                 }
             }
         }
-        self.fix_nodes(memory, path, key_deleted)?;
+
+        let mut child_memory_usage = 0;
+        for handle in path.into_iter().rev() {
+            // First, recompute memory usage, emulating the recursive descent.
+            let TrieNodeWithSize { node, mut memory_usage } = memory.destroy(handle);
+            memory_usage += child_memory_usage;
+            memory.store_at(handle, TrieNodeWithSize { node, memory_usage });
+
```

Comment:
Aah crap, this along with duplication of `memory.destroy(handle)` in `squash_node` and missing `else` case in `if key_deleted` completely confused me, but after staring at this for 10 min, I think it makes sense